### ISSUE TITLE
Layout regression on windows 10?

### ISSUE THREAD

**Issue by kwonoj** [servo/servo#20505]
I just rebased dev build to top of master, and seeing below screenshots when run `mach run --dev` :
(Windows 10, local dev build from master https://github.com/servo/servo/commit/a208d4246cf0e02a1a22196f26692bd29fb35011)

![image](https://user-images.githubusercontent.com/1210596/38209929-b503452c-366a-11e8-8859-27f769fe0961.png)

It does seems only layout if off, while eventhandler still maps to correct location (i.e clicking `Download`..button doesn't work if click where currently button is rendered). I couldn't do enough bisect / or dig into where it comes from (and it may possible my machine's only odd case having issues)

**Comment by jdm** [comment]
I don't see this on macOS.

**Comment by kwonoj** [comment]
So far this seems on win only (my mac / linux works correctly)

**Comment by kwonoj** [comment]
I take it back for regressions around commits - it does seem like dpi-related. This is my first attempt to run other win dev machine which have 150% dpi settings (main machine was on 100%). Changing dpi scaling I got different rendering results, while only 100% shows correct layout. 

Not sure if it's dupe (I couldn't search exact dupes on my end) - if so, feel freely close it. 🙏

100%:
![100](https://user-images.githubusercontent.com/1210596/38219701-b36cbe5c-368b-11e8-824d-a9912bff6816.png)

150%:
<img width="641" alt="150" src="https://user-images.githubusercontent.com/1210596/38219735-dceae83a-368b-11e8-94a5-e2658a92f19e.png">

200%:
![200](https://user-images.githubusercontent.com/1210596/38219738-e03e0332-368b-11e8-8bf0-514bd0adf085.png)

**Comment by kwonoj** [comment]
it may seems upstream related: https://github.com/tomaka/winit/issues/105

**Comment by atouchet** [comment]
I am having issues reproducing this. This is what I am seeing at 150% scaling:

![150% scaling](https://i.imgur.com/WvtBDKQ.png)

**Comment by kwonoj** [comment]
Interesting, curious what could be other factor to make differences.

**Comment by paulrouget** [comment]
@kwonoj can you check if it regressed because of https://github.com/servo/servo/pull/20228 ?

**Comment by paulrouget** [comment]
Or maybe https://github.com/servo/servo/pull/19895

**Comment by kwonoj** [comment]
@paulrouget yes, https://github.com/servo/servo/pull/20228 seems related. I built master at 28c92db26837531e75327cff7727ed8bc1c70b48, right before #20228 and it renders correctly.

**Comment by kwonoj** [comment]
I have few theory and trying to verify now, just taking some time due to sadly slow build speed on windows.

**Comment by FallingSnow** [comment]
I'm on Arch Linux, Gnome 3.28.0, Wayland. My desktop is scaled 200% on a 4k screen. I also experience a scaling issue.

![screenshot from 2018-04-08 08-37-39](https://user-images.githubusercontent.com/2512197/38469358-4649ca50-3b08-11e8-99d9-df4a2a4d736b.png)

**Comment by paulrouget** [comment]
@FallingSnow what happens if you click where the blue button is "supposed" to be? As in, do the coordinates of the click are properly calculated, or are they also scaled up?

**Comment by FallingSnow** [comment]
It would seem the coordinates of the click do not line up with how the page is rendered. Clicking the Download Servo button results in nothing happening.

**Comment by paulrouget** [comment]
If you use X11 instead, do you have the same issue?

**Comment by kwonoj** [comment]
Latest winit have support for hidpi across platform (https://github.com/tomaka/winit/blob/master/CHANGELOG.md#version-0160-2018-06-25) . I tried to bump but realized lot of other dependencies needed to be updated as well, such as servo-skia.

**Comment by kwonoj** [comment]
@paulrouget not directly, it includes cherrypick of my commits with patched skia (https://github.com/servo/servo/pull/21087/files#diff-80398c5faae3c069e4e6aa2ed11b28c0R19), guess that patch made work to remove duplicated dependencies. I was thought of waiting skia bump to pickup latest glutin.

**Comment by atouchet** [comment]
Did #21087 fix this?

**Comment by FallingSnow** [comment]
Just tried latest nightly on X11, Arch Linux, Gnome 3.28.2. Nothing but a grey window now.
![screenshot from 2018-07-04 09-18-57](https://user-images.githubusercontent.com/2512197/42287470-50bd05d6-7f6b-11e8-94ab-0bf632c26b8a.png)

**Comment by FallingSnow** [comment]
Never mind, just much slower at loading than previously. The scaling issue seems to be partially correct on X11 now. Partially because the scaling on gnome is set to 200% but servo is rendering at 100%.
